metacat-connector-snowflake/src/main/java/com/netflix/metacat/connector/snowflake/SnowflakeConnectorTableService.java

```diff
@@ -152,12 +152,16 @@ protected ResultSet getColumns(
         @Nonnull @NonNull final Connection connection,
         @Nonnull @NonNull final QualifiedName name
     ) throws SQLException {
-        return connection.getMetaData().getColumns(
-            connection.getCatalog(),
-            name.getDatabaseName(),
-            name.getTableName(),
-            JdbcConnectorUtils.MULTI_CHARACTER_SEARCH
-        );
+        try {
+            return connection.getMetaData().getColumns(
+                connection.getCatalog(),
+                name.getDatabaseName(),
+                name.getTableName(),
+                JdbcConnectorUtils.MULTI_CHARACTER_SEARCH
+            );
+        } catch (SQLException e) {
+            throw this.exceptionMapper.toConnectorException(e, name);
+        }
```

Comment:
It looks that this is the fix for the getColumns failure with tableNotFound. Another way to do is to check in the method ```public TableInfo get()```.